### PR TITLE
Fix Fetch-Depth Calculation in the review request action.

### PR DESCRIPTION
**Pull request by OhmSpectator**
Ok, now we know that this action does not break anything if fails, so we can continue testing it =)

This PR fixes a bug where direct arithmetic was used in a GitHub Actions expression, which is not supported.

### CODE REVIEW

**Comment by OhmSpectator** [comment]
Ok, now the calculations and review assignment look fine, but I see a permissions error:
```
error fetching organization teams: GraphQL: Resource not accessible by integration (organization.teams)
```
Most probably, it's related to some permissions on GitHub token in use...
@uncleDecart, @dautovri, @eriknordmark, so you have any ideas there?

File: .github/workflows/request_codeowners_review.yml [hunk 1]
```diff
@@ -8,11 +8,16 @@ on:
 jobs:
   auto_request_review:
     runs-on: ubuntu-latest
+    permissions: write-all
     steps:
+      - name: Calculate fetch-depth
+        id: calc
+        run: echo "::set-output name=depth::$(echo $((${{ github.event.pull_request.commits }} + 1)))"
```
Comment (uncleDecart) [inline .github/workflows/request_codeowners_review.yml:16]: ::set-output is [deprecated](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

**Comment by uncleDecart** [comment]
>Ok, now the calculations and review assignment look fine, but I see a permissions error:
`error fetching organization teams: GraphQL: Resource not accessible by integration (organization.teams)`
Most probably, it's related to some permissions on GitHub token in use...

@OhmSpectator try to use `write-all` permission

**Comment by OhmSpectator** [comment]
> > Ok, now the calculations and review assignment look fine, but I see a permissions error:
> > `error fetching organization teams: GraphQL: Resource not accessible by integration (organization.teams)`
> > Most probably, it's related to some permissions on GitHub token in use...
> 
> @OhmSpectator try to use `write-all` permission

That was he first thing I tried... Now, I want to check with less granularity.

File: .github/workflows/request_codeowners_review.yml (hunk 1 above)
Comment (OhmSpectator) [reply .github/workflows/request_codeowners_review.yml:16]: Will fix now, thanks

**Comment by OhmSpectator** [comment]
Nnnah... Still does not work. I guess it's a problem on the Organization Setting level...

**Review by eriknordmark — COMMENTED** [review]
Are you sure the permissions issue isn't due to you running it in your fork/branch/workspace?

I'm open to merging this without the added permissions lines and see what happens in master.

**Comment by OhmSpectator** [comment]
> Are you sure the permissions issue isn't due to you running it in your fork/branch/workspace?

The action runs in this PR as it is introduced here (that's how I tested it).
The result of the run in the PR is the permission error:
https://github.com/lf-edge/eve/actions/runs/6352669616/job/17255983179?pr=3474

**Comment by OhmSpectator** [comment]
Discussed with @yash-zededa a little. We still have to request the corresponding permissions from @lf-edge. @eriknordmark, could you please create a ticket?

**Comment by OhmSpectator** [comment]
Created a request:
https://jira.linuxfoundation.org/plugins/servlet/desk/portal/2/IT-26059

**Comment by OhmSpectator** [comment]
Looks like the pull_requests permissions will not be enough:
https://github.com/cli/cli/issues/7671

**Comment by OhmSpectator** [comment]
Rebased on master

**Comment by uncleDecart** [comment]
@OhmSpectator I see curl [workaround](https://github.com/cli/cli/issues/4844#issuecomment-1534825316)

**Comment by OhmSpectator** [comment]
> @OhmSpectator I see curl [workaround](https://github.com/cli/cli/issues/4844#issuecomment-1534825316)

That's what I'm testing right now

**Comment by OhmSpectator** [comment]
Failed again... Checking...
```
{
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/pulls/review-requests#request-reviewers-for-a-pull-request"
}
```

File: .github/workflows/request_codeowners_review.yml [hunk 2]
```diff
@@ -45,10 +51,21 @@ jobs:
           if [ -n "$reviewers" ]; then
             # Remove @ from reviewers (it stays in the beginning of the username)
             reviewers_cleaned=${reviewers//@/}
-            # Convert reviewers to comma separated list
-            reviewers_comma_separated=$(echo "$reviewers_cleaned" | tr ' ' ',')
-            echo "Requesting review from: $reviewers_comma_separated"
-            gh pr edit ${{ github.event.pull_request.number }} --add-reviewer $reviewers_comma_separated
+            # Convert the space-separated usernames into a Bash array
```
Comment (eriknordmark) [inline .github/workflows/request_codeowners_review.yml:54]: Would it make sense to add a comment with a reference to the reported issue? And/or put that reference in the commit message?

Comment (OhmSpectator) [reply .github/workflows/request_codeowners_review.yml:54]: Yep, I'll link if it makes sense.
But the workflow still fails :(

**Comment by OhmSpectator** [comment]
@eriknordmark, I wanted to check if we can merge this PR. @europaul suggested that the permissions might work differently if we run it from the workflow version that is not yet merged. Even if it still fails, merging it won't break anything as it is already failing (with a different error).

**Review by europaul — APPROVED** [review]
let's give it a try!

**Review by eriknordmark — APPROVED** [review]
LGTM but we might not need the permissions: statement.

**Pull request merged**